platform/windows/windows_utils.cpp

```diff
@@ -181,7 +181,7 @@ Error WindowsUtils::copy_and_rename_pdb(const String &p_dll_path) {
 
 			utf8_name.resize(original_path_size - suffix_size + 1); // +1 for the \0
 			utf8_name[utf8_name.size() - 1] = '\0';
-			new_pdb_base_name.parse_utf8(utf8_name);
+			new_pdb_base_name.parse_utf8(utf8_name, utf8_name.size());
```

Comment:
```suggestion
			utf8_name.resize(original_path_size - suffix_size);
			new_pdb_base_name.parse_utf8(utf8_name, utf8_name.size());
```
If it's using length, we do not need terminating zero anymore.

Reply:
Actually, resize is also unnecessary:

```suggestion
			new_pdb_base_name.parse_utf8(utf8_name, original_path_size - suffix_size);
```

Reply:
To be honest this whole danse seems suspicious. Transforming the whole thing to utf8 to remove _suffix length_ chars and then reparse, can provoke an invalid codepoint if we have multi bytes codepoints, which seems like a very bad idea. Probably way better to just remove _suffix length_ codepoints in utf32, as the result will be correct and the whole idea is to truncate if it's get too long. No allocations, no parsing.

Reply:
This seems a little out of scope for this PR. I will do as you suggested for now, and maybe look into simplifying this in another PR.